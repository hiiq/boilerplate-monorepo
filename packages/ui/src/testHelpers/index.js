import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
