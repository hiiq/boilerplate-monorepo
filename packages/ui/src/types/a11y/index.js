import { Role } from './role';

export const A11y = {
  Role,
};
