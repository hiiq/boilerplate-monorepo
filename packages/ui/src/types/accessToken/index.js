// This variable is the only place that the access token is stored
let token = null;

const clear = () => {
  token = null;
};
const read = () => {
  return token;
};
const set = newToken => {
  token = newToken;
};

export const AccessToken = {
  clear,
  read,
  set,
};
