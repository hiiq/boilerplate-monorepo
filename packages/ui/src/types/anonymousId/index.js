import { LocalStorage } from '@boilerplate-monorepo/ui-common';
import cuid from 'cuid';

const makeId = () => `anonymous-${cuid()}`;
const clear = () => localStorage.removeItem(LocalStorage.ANONYMOUS_ID);

const set = () => {
  const newId = makeId();

  localStorage.setItem(LocalStorage.ANONYMOUS_ID, newId);

  return newId;
};

const read = () => localStorage.getItem(LocalStorage.ANONYMOUS_ID) || set();

export const AnonymousId = {
  clear,
  read,
  set,
};
