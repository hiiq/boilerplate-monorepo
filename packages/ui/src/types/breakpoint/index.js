export const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};
