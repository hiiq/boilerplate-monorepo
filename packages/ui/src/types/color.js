export const Color = {
  black: '#212121',
  border: '#DCDCDC',
  error: '#DB0000',
  focusRing: '#7AACFE',
  muted: '#8D8D8D',
  primary: '#6d4baa',
  success: '#0C9C0C',
  warn: '#DFDF00',
  white: '#FFFFFF',
};
