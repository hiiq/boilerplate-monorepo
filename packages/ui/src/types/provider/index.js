import { Context } from '../context';

const Provider = {
  AUTH: Context.AUTH.Provider,
  FEATURE_FLAGS: Context.FEATURE_FLAGS.Provider,
  INTERNET_CONNECTIVITY: Context.INTERNET_CONNECTIVITY.Provider,
  MODALITY: Context.MODALITY.Provider,
  THEME_SWITCHER: Context.THEME_SWITCHER.Provider,
};

export { Provider };
