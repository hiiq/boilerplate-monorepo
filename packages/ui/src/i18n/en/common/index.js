import { Site } from 'types/site';

export const common = {
  about: 'About',
  dashboard: 'Dashboard',
  footer: `©️ {{fullYear}} ${Site.name}. All Rights reserved.`,
  home: 'Home',
  loading: 'Loading',
  loadingScreenReader: 'Please wait while content is being loaded',
  siteName: `${Site.name} :: {{title}}`,
};
