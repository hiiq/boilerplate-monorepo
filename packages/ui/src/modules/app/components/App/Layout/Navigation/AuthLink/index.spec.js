import { User } from '@boilerplate-monorepo/common';
import React from 'react';
import { render } from 'testHelpers';
import { Provider } from 'types/provider';
import { AuthLink } from '.';

jest.mock('../NavLink', () => ({
  NavLink: props => <x-NavLink {...props} />,
}));

describe('<AuthLink/>', () => {
  const defaultProps = {
    onClick: jest.fn().mockName('onClick'),
  };

  const baseAuthContext = {
    login: jest.fn().mockName('login'),
    logout: jest.fn().mockName('logout'),
    user: User.uiExample({
      email: 'EMAIL',
    }),
  };

  const renderComponent = authContext => overrides => {
    // eslint-disable-next-line react/prop-types
    const wrapper = ({ children }) => (
      <Provider.AUTH value={authContext}>{children}</Provider.AUTH>
    );

    return render(<AuthLink {...defaultProps} {...overrides} />, { wrapper });
  };

  test('renders properly', () => {
    const authContext = {
      ...baseAuthContext,
      isAuthenticated: true,
    };

    const { container } = renderComponent(authContext)();

    expect(container).toMatchSnapshot();
  });
});
