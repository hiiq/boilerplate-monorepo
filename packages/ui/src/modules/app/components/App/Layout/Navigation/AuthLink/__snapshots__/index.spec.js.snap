// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLink/> renders properly 1`] = `
<div>
  <x-navlink
    classname="AuthLink__StyledNavLink-gt2cja-0 VDgZX"
    route="[object Object]"
  />
</div>
`;
